qt/scientific_interfaces/Indirect/InelasticDataManipulationElwinTabView.cpp

```diff
@@ -0,0 +1,537 @@
+  m_elwTree->addProperty(m_properties["Normalise"]);
+
+  // We always want one range selector... the second one can be controlled from
+  // within the elwinTwoRanges(bool state) function
+  auto integrationRangeSelector = m_uiForm.ppPlot->addRangeSelector("ElwinIntegrationRange");
+  integrationRangeSelector->setBounds(-DBL_MAX, DBL_MAX);
+  connect(integrationRangeSelector, SIGNAL(minValueChanged(double)), this, SLOT(minChanged(double)));
+  connect(integrationRangeSelector, SIGNAL(maxValueChanged(double)), this, SLOT(maxChanged(double)));
+  // create the second range
+  auto backgroundRangeSelector = m_uiForm.ppPlot->addRangeSelector("ElwinBackgroundRange");
+  backgroundRangeSelector->setColour(Qt::darkGreen); // dark green for background
+  backgroundRangeSelector->setBounds(-DBL_MAX, DBL_MAX);
+  connect(integrationRangeSelector, SIGNAL(selectionChanged(double, double)), backgroundRangeSelector,
+          SLOT(setRange(double, double)));
+  connect(backgroundRangeSelector, SIGNAL(minValueChanged(double)), this, SLOT(minChanged(double)));
+  connect(backgroundRangeSelector, SIGNAL(maxValueChanged(double)), this, SLOT(maxChanged(double)));
+
+  connect(m_dblManager, SIGNAL(valueChanged(QtProperty *, double)), this, SLOT(updateRS(QtProperty *, double)));
+  connect(m_dblManager, SIGNAL(valueChanged(QtProperty *, double)), this, SIGNAL(valueChanged(QtProperty *, double)));
+  connect(m_blnManager, SIGNAL(valueChanged(QtProperty *, bool)), this, SLOT(twoRanges(QtProperty *, bool)));
+  connect(m_blnManager, SIGNAL(valueChanged(QtProperty *, bool)), this, SIGNAL(valueChanged(QtProperty *, bool)));
+  twoRanges(m_properties["BackgroundSubtraction"], false);
+
+  connect(m_uiForm.dsInputFiles, SIGNAL(filesFound()), this, SIGNAL(filesFound()));
+  connect(m_uiForm.cbPreviewFile, SIGNAL(currentIndexChanged(int)), this, SIGNAL(previewIndexChanged(int)));
+  connect(m_uiForm.spPlotSpectrum, SIGNAL(valueChanged(int)), this, SIGNAL(selectedSpectrumChanged(int)));
+  connect(m_uiForm.cbPlotSpectrum, SIGNAL(currentIndexChanged(int)), this, SIGNAL(selectedSpectrumChanged(int)));
+
+  // Handle plot and save
+  connect(m_uiForm.pbRun, SIGNAL(clicked()), this, SIGNAL(runClicked()));
+  connect(m_uiForm.pbSave, SIGNAL(clicked()), this, SIGNAL(saveClicked()));
+  connect(m_uiForm.pbPlotPreview, SIGNAL(clicked()), this, SIGNAL(plotPreviewClicked()));
+
+  // Set any default values
+  m_dblManager->setValue(m_properties["IntegrationStart"], -0.02);
+  m_dblManager->setValue(m_properties["IntegrationEnd"], 0.02);
+
+  m_dblManager->setValue(m_properties["BackgroundStart"], -0.24);
+  m_dblManager->setValue(m_properties["BackgroundEnd"], -0.22);
+}
+
+IndirectPlotOptionsView *InelasticDataManipulationElwinTabView::getPlotOptions() { return m_uiForm.ipoPlotOptions; }
+
+QTableWidget *InelasticDataManipulationElwinTabView::getDataTable() const { return m_uiForm.tbElwinData; }
+
+bool InelasticDataManipulationElwinTabView::validate() {
```

Comment:
There's quite a lot of GUI logic going on in this view. This sort of thing is better suited to the presenter with calls into the view to get the values you need to do the checking. 

I'm not going to put comments on them all, but I think this view needs going through to pull out more of that logic. There's a lot of `if` statements that are logic paths that should be tested but can't be when they're in the view. Generally, we want that logic in the presenter where we can test it with mocks.